### PR TITLE
SanitizeMessagePeriod: Add Period at the end of sentences

### LINKED ISSUE #9261: Put a period at the end of messages
## Description

or don't, right now there is no period

**Reproduction**

press T
write "This is a finished sentence"
exhibit no dot
despair

**Additional context**

muh MRP

### PR DESCRIPTION
**Pull request by tomasalves8**
## About the PR 
- Creates method ``SanitizeMessagePeriod`` that adds a period at the end of sentences.

Closes #9261

**Changelog**

### CODE REVIEW

**Comment by tomasalves8** [comment]
Dear maintainers, shall you find my worthy of the Junior Contributor role (2nd PR). I would be honored. Tomás#1110

File: Content.Server/Chat/Systems/ChatSystem.cs [hunk 1]
```diff
@@ -469,6 +470,16 @@ private string SanitizeMessageCapital(EntityUid source, string message)
         return message;
     }
 
+    private string SanitizeMessagePeriod(EntityUid source, string message)
```
Comment (Visne) [inline Content.Server/Chat/Systems/ChatSystem.cs:473]: What do you need `source` for?

File: Content.Server/Chat/Systems/ChatSystem.cs [hunk 2]
```diff
@@ -469,6 +470,16 @@ private string SanitizeMessageCapital(EntityUid source, string message)
...
+    {
+        if (string.IsNullOrEmpty(message))
+            return message;
+        // Add period if there isn't one
+        if (message[^1] != '.')
+            message += ".";
```
Comment (Visne) [inline Content.Server/Chat/Systems/ChatSystem.cs:479]: This would also add a period to messages ending with `!` or `?` etc.

**Review by Visne — CHANGES_REQUESTED** [review]

File: Content.Server/Chat/Systems/ChatSystem.cs (hunk 1 above)
Comment (tomasalves8) [reply Content.Server/Chat/Systems/ChatSystem.cs:473]: Yeah on ```SanitizeMessageCapital``` it's the same thing for some reason. I'll remove it.

**Comment by Visne** [comment]
I think the best way to go about this is by never adding a period if the last character is not an alphabetical character (a-z). That is assuming this is something we want in the first place (see https://github.com/space-wizards/space-station-14/issues/9261#issuecomment-1221188674).

**Comment by tomasalves8** [comment]
> I think the best way to go about this is by never adding a period if the last character is not an alphabetical character (a-z). That is assuming this is something we want in the first place (see [#9261 (comment)](https://github.com/space-wizards/space-station-14/issues/9261#issuecomment-1221188674)).

Can you give an example where you believe there should be no ponctuation?

**Comment by Visne** [comment]
> Can you give an example where you believe there should be no ponctuation?

`"Hi there."` should not get another period.
`(Hi there.)` should not get another period.
`Here's the thing:` should not get another period.
`**YES**` and many variations should not get a period I think, although it might not be so common.
`Hi…` should not get another period (note that … is a single character).

It's not hard to imagine that there are many other exceptions, and to me it seems like not adding a period when the sentence ends with any character that isn't a-z is the most safe.

**Comment by tomasalves8** [comment]
Something like this?

**Comment by vulppine** [comment]
This behavior should probably be tied to a cvar, so servers can choose if they want to have automatic punctuation or not.

File: Content.Shared/CCVar/CCVars.cs [hunk 3]
```diff
@@ -1045,6 +1045,12 @@ public sealed class CCVars : CVars
         public static readonly CVarDef<bool> FlavorText =
             CVarDef.Create("ic.flavor_text", false, CVar.SERVER | CVar.REPLICATED);
 
+        /// <summary>
+        /// Adds a period at the end of a sentence if the sentence ends in a letter.
+        /// </summary>
+        public static readonly CVarDef<bool> ChatPunctuation =
+            CVarDef.Create("ic.punctuation", false, CVar.SERVER | CVar.REPLICATED);
```
Comment (Visne) [inline Content.Shared/CCVar/CCVars.cs:1052]:
```suggestion
            CVarDef.Create("ic.punctuation", false, CVar.SERVER);
```

I'm pretty sure this doesn't have to be replicated, as the CVar is only written to or read server side.

**Review by Visne — CHANGES_REQUESTED** [review]

**Review by Visne — APPROVED** [review]

**Comment by wrexbe** [comment]
Not a fan.

- It's just as possible an exclamation, a question, or nothing would the appropriate thing at the end.
- It's unnecessary fluff, the period exists to separate sentences, but since each chat message is already separated, it comes across as weird. Which is why it's uncommon to see it in text messages, or on discord.
- It can changes the tone of the message, and may make it seem unfriendly
Ref: The full first page of google results say not to do it: https://www.google.com/search?q=periods+in+chat+messages

**Comment by Visne** [comment]
Good arguments, I agree. But do you think that even with a CVar which is off by default it shouldn't be merged? Even taking into consideration you arguments, I can still imagine this feature being useful for HRP servers that want to enforce a serious tone.

**Comment by metalgearsloth** [comment]
Are there servers that actively do this? At least in the case of detailed examine there was a precedent for it on some RP servers even if we left it turned off.

**Comment by Visne** [comment]
I don't know, I don't play SS13 because it sucks.

**Comment by Just-a-Unity-Dev** [comment]
> Are there servers that actively do this? At least in the case of detailed examine there was a precedent for it on some RP servers even if we left it turned off.

IIRC Bay-downstreams do this such as that one SCP fork Foundation 19

**Comment by rbertoche** [comment]
if you make a question in english without question marker, people will get you. add a period to that, and I'm not really sure, will it work out well.
not sure how and why that's not often mentioned, it seems a real conversation breaker literally
not sure how periods make things more serious, seems just more dry to me

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**